src/core/thread/mle_router.cpp

```diff
@@ -2943,6 +2948,28 @@ ThreadError MleRouter::GetChildInfoByIndex(uint8_t aChildIndex, otChildInfo &aCh
     return error;
 }
 
+ThreadError MleRouter::AddChild(const otChildInfo &aChildInfo)
+{
+    ThreadError error = kThreadError_None;
+    Child *child;
+
+    VerifyOrExit((child = NewChild()) != NULL, error = kThreadError_NoBufs);
+
+    memset(child, 0, sizeof(*child));
+
+    memcpy(&child->mMacAddr, &aChildInfo.mExtAddress, sizeof(child->mMacAddr));
+    child->mValid.mRloc16 = aChildInfo.mRloc16;
+    child->mTimeout = aChildInfo.mTimeout;
+    child->mMode = aChildInfo.mRxOnWhenIdle | aChildInfo.mSecureDataRequest | aChildInfo.mFullFunction |
+                   aChildInfo.mFullNetworkData;
+    child->mNetworkDataVersion = aChildInfo.mNetworkDataVersion;
+    child->mState = Neighbor::kStateValid;
```

Comment:
I think we should have a new state here (e.g. `kStateRestored`).  After a router reboots, it will need to send an MLE Child Update Request message in order to request information from the child (e.g. to retrieve IPv6 address registrations).